packages/dev/core/src/Meshes/geometry.ts

```diff
@@ -600,7 +599,7 @@ export class Geometry implements IGetSetVerticesData {
         this._totalVertices = totalVertices;
         this._totalIndices = totalIndices;
 
-        indexBuffer.is32Bits ||= this._totalIndices > 65535;
+        indexBuffer.is32Bits ||= this._totalVertices > 65535;

```

Comment:
gosh...how is that still here????

Reply:
It looks like we are not relying on it internally

Reply:
My bug (#14551)! We don't use it internally indeed, I added the method to simplify the use case of a forum user (see the link in my first post).

Reply:
ahah we reviewed it so lets share the bug ;-)

![image](https://github.com/user-attachments/assets/8e04c590-0cb6-463f-9b42-8f5dd4974cbb)

Reply:
I think this check is still not quite right. We should use the `AreIndices32Bits` function in `bufferUtils.ts`

Reply:
Although now that I look at the function more. It also has a check for the number of indices which is not good.

Reply:
We can't do the same thing than in `AreIndices32Bits` because the index buffer is a `DataBuffer`, which is a GPU buffer, not an array. We can't retrieve the values from this buffer.

Reply:
I see, but checking total vertices isn't correct either. The indices may or may not be 32bits even if the totalVertices is greater than 64k? What am I missing?

Reply:
Yes, but I think it's the best we can do, and it's conservative (ie. it won't crash): we may create a 32 bits buffer even if there isn't any vertex index > 65535, but we will never create a 16 bits buffer if we have som indices > 65535.

Reply:
Sorry, I don't understand. If the underlying data buffer is a Uint16Array uploaded to the GPU, and we draw it using UNSIGNED_INT, wouldn't that break?

Reply:
You're right, we can't deduce the value correctly in the general case, the user should provide it. I created a PR to fix a number of problems related to index buffer management: #16235